kymatio/scattering1d/scattering1d.py

```diff
@@ -178,8 +178,7 @@ class Scattering1D(object):
         Required precision for the periodization. Defaults to `1e-7`.
     max_order : int, optional
         The maximum order of scattering coefficients to compute. Must be one
-        `1` or `2`. This parameter may be modified after object creation.
-        Defaults to `2`.
+        `1` or `2`. Defaults to `2`.
```

Comment:
This should be in a separate PR (or at least a separate commit).

Reply:
ok. I'll revert this and make a separate PR.
I'd like to think of `max_order` as an immutable and then maybe in the future we could have `orders` as mutable, which defaults to `range(1+max_order)` and needs to be a subset thereof. this is why i removed this sentence: i did not want to offer the guarantee that `max_order` will remain mutable

Reply:
That makes sense. It should be tied into the filter generation, after all.

Not here, but we may want to rename this to `_max_order` so people won't mess with it, but I'm not sure how these things are normally done. There's no encapsulation in Python, right?

Reply:
that's right. nothing is private in Python, but naming an object attribute with a leading underscore is widespread convention for scaring away the user to change these values after construction. the same can be said of methods
(and for C numba bindings, the convention is *two* leading underscores! :D )

Reply:
Ok.